systemtest/edu/washington/escience/myria/systemtest/JsonQuerySubmitTest.java

```diff
@@ -205,58 +210,66 @@ public void jsonQuerySubmitTest() throws Exception {
     assertEquals(((SubQueryEncoding) status.plan).fragments.size(), 3);
   }
 
-  @Test
-  public void joinChainResultTest() throws Exception {
-    // DeploymentUtils.ensureMasterStart("localhost", masterDaemonPort);
-
-    File ingestA0 = new File("./jsonQueries/multiIDB_jwang/ingest_a0.json");
-    File ingestB0 = new File("./jsonQueries/multiIDB_jwang/ingest_b0.json");
-    File ingestC0 = new File("./jsonQueries/multiIDB_jwang/ingest_c0.json");
-    File ingestR = new File("./jsonQueries/multiIDB_jwang/ingest_r.json");
-
-    HttpURLConnection conn;
-    conn = JsonAPIUtils.ingestData("localhost", masterDaemonPort, ingestA0);
-    if (null != conn.getErrorStream()) {
-      throw new IllegalStateException(getContents(conn));
+  public void jsonQuerySubmitWithResultCheck(final String[] ingests, final String[] queries, final String[] expected,
+      final String[] stored) throws Exception {
+    Preconditions.checkArgument(expected.length == stored.length);
+    for (String ingest : ingests) {
+      File ingestFile = new File(ingest);
+      HttpURLConnection conn = JsonAPIUtils.ingestData("localhost", masterDaemonPort, ingestFile);
+      if (null != conn.getErrorStream()) {
```

Comment:
I know you didn't introduce this, but I don't think the constant-first comparison idiom is one we want to encourage.